### PR TITLE
[RFC] Separate parts of the sw renderer into their own functions

### PR DESCRIPTION
**Pull request by Subv**
This helps the readability of the sw renderer since it is no longer a monolithic function.

I want to hear opinions about this change, is it a good idea? is it completely misled?
Hopefully in the future, this will help optimizing the rasterizer into a series of precomputed function calls, doing away with all the redundant switches.

Currently only the tev stages, the alpha test and the fog combiner are factored out.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/2772)

### CODE REVIEW

File: src/video_core/swrasterizer/rasterizer.cpp [hunk 1]
```diff
@@ -114,6 +114,146 @@ static std::tuple<float24, float24, PAddr> ConvertCubeCoord(float24 u, float24 v
     return std::make_tuple(x / z * half + half, y / z * half + half, addr);
 }
 
+Math::Vec4<u8> GetNextCombinerBuffer(unsigned tev_stage_index, const Math::Vec4<u8>& combiner_output,
```
Comment (lioncash) [inline src/video_core/swrasterizer/rasterizer.cpp]: Most of these new functions should be static, no?

**Comment by yuriks** [comment]
Move these to texturing.cpp (I chose a bad name for that one... should be combiners.cpp or something) where some of this code already is.

**Comment by yuriks** [comment]
(And re. if this is a good idea or not: it totally is)

File: src/video_core/swrasterizer/rasterizer.cpp [hunk 2]
```diff
@@ -410,6 +550,7 @@ static void ProcessTriangleInternal(const Vertex& v0, const Vertex& v1, const Ve
             for (unsigned tev_stage_index = 0; tev_stage_index < tev_stages.size();
                  ++tev_stage_index) {
                 const auto& tev_stage = tev_stages[tev_stage_index];
+
```
Comment (jroweboy) [inline src/video_core/swrasterizer/rasterizer.cpp]: random space added?

**Comment by Subv** [comment]
I'm sorry but I have lost motivation to work on this anymore, I'm closing it so if anyone wants to take it over, feel free.

**Pull request closed without merging**